src/core/instance/proxy.js

```diff
@@ -15,9 +15,11 @@ if (process.env.NODE_ENV !== 'production') {
 
   const warnNonPresent = (target, key) => {
     warn(
-      `Property or method "${key}" is not defined on the instance but ` +
-      `referenced during render. Make sure to declare reactive data ` +
-      `properties in the data option.`,
+      `Property or method "${key}" is not defined on the instance but
+      referenced during render. Make sure that this property is reactive,
+      either in the data option, or for class-based components, by
+      initializing the property.
+      See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.`,
       target
```

Comment:
We need to use concatenations for each line to avoid the whitespace between the lines.